ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/map.cesium.js

```diff
@@ -311,24 +311,50 @@ module.exports = function CesiumMap(
       map.scene.camera.moveEnd.addEventListener(callback)
     },
     doPanZoom: function(coords) {
-      const cartArray = coords.map(function(coord) {
-        return Cesium.Cartographic.fromDegrees(
-          coord[0],
-          coord[1],
-          map.camera._positionCartographic.height
-        )
-      })
-      if (cartArray.length === 1) {
-        const point = Cesium.Ellipsoid.WGS84.cartographicToCartesian(
-          cartArray[0]
-        )
-        this.panToCoordinate(point, 2.0)
-      } else {
-        const rectangle = Cesium.Rectangle.fromCartographicArray(cartArray)
+      if (Array.isArray(coords[0][0])) {
+        const rectangles = []
+        coords.forEach(function(polygon) {
+          const cartArray = polygon.map(function(coord) {
+            return Cesium.Cartographic.fromDegrees(
+              coord[0],
+              coord[1],
+              map.camera._positionCartographic.height
+            )
+          })
+          const rectangle = Cesium.Rectangle.fromCartographicArray(cartArray)
+          rectangles.push(rectangle)
+        })
+        let rectangle = rectangles[0]
+        let currentArea = 0
+        rectangles.forEach(function(r) {
+          if (Math.abs((r.east - r.west) * (r.south - r.north)) > currentArea) {
```

Comment:
Should you extract this if logic into a function that's more clear?

Reply:
The logic in this function is being moved to the gazetteer file